[BUG] Cython 3.0.0b1 not handling -std=c++03

### Describe the bug

Making progress on porting yt to Cython 3.0, but I'm still encountering one compilation error where what _seems_ to be happening is that Cython produces C++11-only code despite an explicit directive
`# distutils: extra_compile_args = ['-std=c++03']`

The following errors are outputted
```
yt/utilities/lib/cykdtree/kdtree.cpp:573:81: error: default template arguments may not be used in function templates without ‘-std=c++11’ or ‘-std=gnu++11’
  573 |     template<typename U=T> bool operator==(const __Pyx_FakeReference<U>& other) const { return *ptr == *other.ptr; }
      |                                                                                 ^~~~~
yt/utilities/lib/cykdtree/kdtree.cpp:574:81: error: default template arguments may not be used in function templates without ‘-std=c++11’ or ‘-std=gnu++11’
  574 |     template<typename U=T> bool operator!=(const __Pyx_FakeReference<U>& other) const { return *ptr != *other.ptr; }
```

I have not been able to reproduce this locally, so all I have to show is a failed build on GHA, see
- [context](https://github.com/yt-project/yt/pull/4374#issuecomment-1468208388)
- [failed build](https://github.com/yt-project/yt/actions/runs/4416612698/jobs/7741148482)


This could result from an intentional change (support for C++ older than 11 dropped ...), however I haven't been able to find any mention of it in the migration guide or elsewhere. If the failure is intended, maybe the error message could make it clearer too.


### Code to reproduce the behaviour:

_No response_

### Expected behaviour

_No response_

### OS

Linux

### Python version

3.11.2

### Cython version

3.0.0b1

### Additional context

_No response_

--- a/Cython/Utility/ModuleSetupCode.c
+++ b/Cython/Utility/ModuleSetupCode.c
@@ -603,8 +603,8 @@ class __Pyx_FakeReference {
     // TODO(robertwb): Delegate all operators (or auto-generate unwrapping code where needed).
     template<typename U> bool operator ==(const U& other) const { return *ptr == other; }
     template<typename U> bool operator !=(const U& other) const { return *ptr != other; }
-    template<typename U=T> bool operator==(const __Pyx_FakeReference<U>& other) const { return *ptr == *other.ptr; }
-    template<typename U=T> bool operator!=(const __Pyx_FakeReference<U>& other) const { return *ptr != *other.ptr; }
+    template<typename U> bool operator==(const __Pyx_FakeReference<U>& other) const { return *ptr == *other.ptr; }
+    template<typename U> bool operator!=(const __Pyx_FakeReference<U>& other) const { return *ptr != *other.ptr; }
   private:
     T *ptr;
 };

